stacks_queues/queue.py

```diff
@@ -15,47 +14,85 @@ def __init__(self):
         self.front = -1
         self.rear = -1
         self.size = 0
-      
 
     def enqueue(self, element):
         """ Adds an element to the Queue
             Raises a QueueFullException if all elements
             In the store are occupied
             returns None
         """
-        pass
+
+        if ((self.rear + 1) % self.buffer_size == self.front):
+            raise QueueFullException()
+        elif self.empty():
+            self.front = 0
+            self.rear = 0
+            self.store[self.rear] = element
+        elif ( (self.rear == self.buffer_size - 1) and (self.front != 0) ):
+            self.rear = 0
+            self.store[self.rear] = element
+        else:
+            self.rear += 1
+            self.store[self.rear] = element
 
     def dequeue(self):
         """ Removes and returns an element from the Queue
             Raises a QueueEmptyException if 
             The Queue is empty.
         """
-        pass
+
+        if self.empty():
+            raise QueueEmptyException()
+
+        content = self.store[self.front]
+        self.store[self.front] = None
+
+        if (self.front == self.rear):
+            self.front = -1
+            self.rear = -1
+        else:
+            self.front = (self.front + 1) % self.buffer_size
+        return content
```

Comment:
👍 Nicely done, using this strategy the `enqueue` could be simplified a bit as well.